bindings/wasm/src/lib.rs

```diff
@@ -66,208 +61,564 @@ fn js_error<T: std::fmt::Debug>(e: T) -> JsValue {
-    array.copy_from_slice(bytes); 
-    Hash(array)
+fn create_hash_from_string(bytes: String) -> Result<Hash, JsValue> {
+    let hash = Hash::from_inner_unchecked(
+        TryteBuf::try_from_str(&bytes)
+            .map_err(js_error)?
+            .as_trits()
+            .encode(),
+    );
+    Ok(hash)
+}
+
+fn create_hash(bytes: JsValue) -> Result<Hash, JsValue> {
+    let bytes: String = bytes.into_serde().map_err(js_error)?;
+    create_hash_from_string(bytes)
 }
 
-#[wasm_bindgen]
-impl Client {
-    #[wasm_bindgen(constructor)]
-    pub fn new(uri: &str) -> Result<Client, JsValue> {
-        console_error_panic_hook::set_once();
-        let client = Client {
-            client: iota::Client::new(uri)
-                .map_err(|e| e.to_string())?
-        };
-        Ok(client)
+fn create_hash_array(bytes_vec: JsValue) -> Result<Vec<Hash>, JsValue> {
+    let hashes_vecs: Vec<String> = bytes_vec.into_serde().map_err(js_error)?;
+    let mut hashes = Vec::new();
+    for hash in hashes_vecs {
+        hashes.push(create_hash_from_string(hash)?);
     }
+    Ok(hashes)
+}
 
-    #[wasm_bindgen(js_name = "getNodeInfo")]
-    pub async fn get_node_info(self) -> Result<JsValue, JsValue> {
-        let node_info = self.client.get_node_info()
-            .await
-            .map_err(|e| JsValue::from(e.to_string()))?;
-        let res = response_to_js_value(node_info)?;
-        Ok(res)
+fn create_address(address: String) -> Result<Address, JsValue> {
+    let address = Address::from_inner_unchecked(
+        TryteBuf::try_from_str(&address)
+        .map_err(js_error)?
+        .as_trits()
+        .encode(),
+    );
+    Ok(address)
+}
+
+fn create_addresses(addresses: JsValue) -> Result<Vec<Address>, JsValue> {
+    let addresses_vec: Vec<String> = addresses.into_serde().map_err(js_error)?;
+    let mut addresses = Vec::new();
+    for address in addresses_vec {
+        let address = create_address(address)?;
+        addresses.push(address);
     }
+    Ok(addresses)
+}
 
-    #[wasm_bindgen(js_name = "getNewAddress")]
-    pub async fn get_new_address(self, seed: String, index: Option<u64>, security: Option<u8>) -> Result<JsValue, JsValue> {
-        let encoded_seed = IotaSeed::<Kerl>::from_buf(
-            TryteBuf::try_from_str(&seed)
-            .map_err(js_error)?
-            .as_trits()
-            .encode::<T1B1Buf>(),
-        )
-            .map_err(js_error)?;
+#[wasm_bindgen(js_name = "addNode")]
+pub fn add_node(uri: &str) -> Result<(), JsValue> {
+    console_error_panic_hook::set_once();
+    iota::Client::add_node(uri)
+        .map_err(js_error)?;
+    Ok(())
+}
 
-        let mut builder = self.client.get_new_address();
-        builder = builder.seed(&encoded_seed);
+#[wasm_bindgen(js_name = "addNeighbors")]
+pub async fn add_neighbors(uris: JsValue) -> Result<JsValue, JsValue> {
+    let uris: Vec<String> = uris.into_serde().map_err(js_error)?;
+    let added_neighbords = iota::Client::add_neighbors(uris)
```

Comment:
```suggestion
    let added_neighbors = iota::Client::add_neighbors(uris)
```

Reply:
You're the official typo finder :D thanks